---
title: "AD Ortholog Exploration"
author: "Alexander Bergendorf"
date: "2023-08-07"
output: html_document
---


# Library Import
```{r}
library(Seurat)
library(mixOmics)
library(h2o)
library(tidyverse)
library(orthogene)
library(dittoSeq)
library(openxlsx)
home_dir <- "C:/Users/aberg/OneDrive/Desktop/Brubaker Summer Research 2023"
```

# PCA Exploration
## Loading Mouse Data
```{r}
combined <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_adwt_labeled.RData"))
combined <- subset(combined, subset = summary == "Microglia")
```

## Run PCA to isolate mouse Microglia
```{r}
DefaultAssay(combined) <- "RNA"
combined <- NormalizeData(combined)
combined <- FindVariableFeatures(combined, selection.method = "vst", nfeatures = 3000)
all.genes <- rownames(combined)
combined <- ScaleData(combined, features = all.genes)
combined <- RunPCA(combined, features = VariableFeatures(object = combined))
```


```{r}
dittoDimPlot(combined, var = "Genotype", reduction.use = "pca")
```

```{r}
DimHeatmap(combined, dims = 1:4)
```

A lot of these principle components are separated by only a few cells... I think we need to cluster the cells again to get a better representation 

```{r}
combined <- FindNeighbors(combined, dims = 1:5)
combined <- FindClusters(combined, resolution = 0.5)
combined <- RunUMAP(combined, dims = 1:5)
```


```{r}
dittoDimPlot(combined, var = "Genotype", reduction = "umap")
dittoDimPlot(combined, "seurat_clusters", do.label = TRUE, legend.show = FALSE, labels.size = 3) 
dittoDimPlot(combined, var = "CellType", reduction = "umap")
```

```{r}
markers <- FindAllMarkers(combined, assay = "RNA", only.pos = TRUE, logfc.threshold = 0.5, test.use = "MAST", latent.vars = "orig.ident") 

clust_counts <- table(combined@meta.data[,"seurat_clusters"]) 

clust_markers_list <- list() 

for(i in unique(markers$cluster)){ 

clustname <- paste0('cluster', i, '_N', clust_counts[i]) 

clust_markers_list[[clustname]] <- subset(markers, markers$cluster == i) 

rownames(clust_markers_list[[clustname]]) <- clust_markers_list[[clustname]]$gene 

} 

write.xlsx(clust_markers_list, file = paste0(home_dir, "/mouse_microglia_reclustering.xlsx"))
```

```{r}
# Find sub cluster of cluster 2 which has some microglial and some non-microglial cells
combined <- FindSubCluster(
  combined,
  cluster = 2,
  graph.name = "RNA_snn",
  subcluster.name = "sub.cluster",
  resolution = 2,
  algorithm = 1
)
```


```{r}
# Cluster 2_1, 3, 4, and 6 are all non-microglial cels
combined <- subset(combined, subset = sub.cluster %in% c("2_1","3", "4", "6"), invert = TRUE)

```

```{r}
# Here we can see all mouse microglial cells
dittoDimPlot(combined, "sub.cluster", do.label = TRUE, legend.show = FALSE, labels.size = 3)
```




```{r}
# Save barcodes of good microglial cells for future use
good_microglia <- Cells(combined)
#saveRDS(good_microglia, file = paste0(home_dir, "/AD_mouse_objects_ortholog/good_microglia_barcodes.RData"))
```


## Rerun PCA mouse
```{r}
# Rerunning PCA after filtering
DefaultAssay(combined) <- "RNA"
combined <- NormalizeData(combined)
combined <- FindVariableFeatures(combined, selection.method = "vst", nfeatures = 3000)
all.genes <- rownames(combined)
combined <- ScaleData(combined, features = all.genes)
combined <- RunPCA(combined, features = VariableFeatures(object = combined))
```
```{r}
# Variances of mosue PCs
stdev <- combined@reductions$pca@stdev
variance <- stdev^2
variance <- variance / sum(variance)
variance
write.csv(variance, paste0(home_dir, "/pca_variances_mouse.csv"))
```


## Figure 2B
```{r}
# Cluster cells and visualize on UMAP for paper visualization
combined <- FindNeighbors(combined, dims = 1:5)
combined <- FindClusters(combined, resolution = 0.5)
combined <- RunUMAP(combined, dims = 1:5)
```


```{r}
p1 <- dittoDimPlot(combined, "Genotype", color.panel = c("#F8766D", "#00BFC4"))
cairo_ps(filename= paste0(home_dir, "/Figures/Figure 2/2B.eps"), width = 9, height = 7) # name the file as an EPS
p1 #variable of the figure that is defined/saved, can be used any name
dev.off()
```

```{r}
# Counts of mouse microglial cells
count_table <- table(combined@meta.data$Genotype)
write.csv(count_table, file = paste0(home_dir, "/Figures/Figure 2/mouse_counts.csv"))
```


```{r}
ElbowPlot(combined, ndims = 50)
```


```{r}
# We can see here that PC_2 separates mouse AD and WT cells.
DimPlot(combined, group.by = "Genotype",dims = c(1, 2), reduction = "pca")
```


```{r}
# Pull principal components from Seurat object
principle_components <- combined@reductions$pca@feature.loadings
head(principle_components)
principle_components <- as.data.frame(principle_components)
```

Now that we have the eigenvectors/eigenvalues of the PCA covariance matrix, lets bring in the human data and project the data into mouse PC space

## Run PCA to isolate human microglia
```{r}
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_adwt_labeled_ortholog.RData"))
human <- subset(human, subset = summary == "Microglia")
```


```{r}
# Run initial PCA
DefaultAssay(human) <- "RNA"
human <- NormalizeData(human)
human <- FindVariableFeatures(human, selection.method = "vst", nfeatures = 3000)
all.genes <- rownames(human)
human <- ScaleData(human, features = all.genes)
human <- RunPCA(human, features = VariableFeatures(object = human))
```

```{r}
ElbowPlot(human)
```

```{r}
# Same case as for Mouse, few cells causing separation, so, let's filter non-microglial cells out
dittoDimPlot(human, var = "Genotype", reduction.use = "pca")
```

```{r}
# Clustering 
human <- FindNeighbors(human, dims = 1:10)
human <- FindClusters(human, resolution = 0.5)
human <- RunUMAP(human, dims = 1:10)
```

```{r}
dittoDimPlot(human, var = "Genotype", reduction = "umap")
dittoDimPlot(human, "seurat_clusters", do.label = TRUE, legend.show = FALSE, labels.size = 3) 
```

```{r}
markers <- FindAllMarkers(human, assay = "RNA", only.pos = TRUE, logfc.threshold = 0.5, test.use = "MAST", latent.vars = "orig.ident") 

clust_counts <- table(human@meta.data[,"seurat_clusters"]) 

clust_markers_list <- list() 

for(i in unique(markers$cluster)){ 

clustname <- paste0('cluster', i, '_N', clust_counts[i]) 

clust_markers_list[[clustname]] <- subset(markers, markers$cluster == i) 

rownames(clust_markers_list[[clustname]]) <- clust_markers_list[[clustname]]$gene 

} 

write.xlsx(clust_markers_list, file = paste0(home_dir, "/human_microglia_reclustering.xlsx"))
```


```{r}
# Based on the markers, 6, 7, 9, 10, and 11 are not microglial cells
human <- subset(human, subset = seurat_clusters %in% c(6, 7, 9, 10, 11), invert = TRUE)
```

```{r}
dittoDimPlot(human, var = "Genotype", reduction = "umap")
dittoDimPlot(human, "seurat_clusters", do.label = TRUE, legend.show = FALSE, labels.size = 3) 
```


```{r}
# Saving barcodes of good human microglia
good_microglia <- Cells(human)
#saveRDS(good_microglia, file = paste0(home_dir, "/AD_human_objects_ortholog/good_microglia_barcodes.RData"))
```



## Get Human GEX matrix
```{r}
# Rerunning PCA again
DefaultAssay(human) <- "RNA"
human <- NormalizeData(human)
all.genes <- rownames(human)
human <- ScaleData(human, features = all.genes)
human_gex_matrix <- as.data.frame(human@assays$RNA@scale.data)
human <- RunPCA(human, features = VariableFeatures(object = human))
```

## Figure 2D
```{r}
ElbowPlot(human)
```


```{r}
# Plotting on UMAP for visualization
human <- FindNeighbors(human, dims = 1:5)
human <- FindClusters(human, resolution = 0.5)
human <- RunUMAP(human, dims = 1:5)
```

```{r}
p2 <- dittoDimPlot(human, "Genotype", color.panel = c("#F8766D", "#00BFC4"))
cairo_ps(filename= paste0(home_dir, "/Figures/Figure 2/2D.eps"), width = 9, height = 7) # name the file as an EPS
p2 #variable of the figure that is defined/saved, can be used any name
dev.off()
```

```{r}
# Count of human microglial cells
count_table <- table(human@meta.data$Genotype)
write.csv(count_table, file = paste0(home_dir, "/Figures/Figure 2/human_counts.csv"))
```


```{r}
# Confirming only orthologs in GEX matrix
ortholog_mat <- convert_orthologs(gene_df = row.names(human_gex_matrix),
                                    input_species = "human",
                                    output_species = "mouse",
                                    non121_strategy = "drop_both_species")

# Filtering the orthologs based on the 3000 principal component loadings
ortholog_mat <- ortholog_mat %>%
  filter(row.names(ortholog_mat) %in% row.names(principle_components))
# Filtering the human gex matrix based on the 3000 principal component loadings
human_gex_matrix <- human_gex_matrix %>%
  filter(row.names(human_gex_matrix) %in% ortholog_mat$input_gene)
# Ordering human_gex_matrix in the same manner as the pc loadings
row.names(human_gex_matrix) <- row.names(ortholog_mat)

# Arrange alphabetically 
principle_components  <- principle_components %>% arrange(row.names(principle_components))
# Arrange alphabetically
human_gex_matrix <- human_gex_matrix %>%
  arrange(row.names(human_gex_matrix))
```

```{r}
# Save mouse object with pca loadings
saveRDS(combined, file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_pca_loadings.RData"))
```


```{r}
# converting to matrix
principle_components <- as.matrix(principle_components)
human_gex_matrix <- as.matrix(human_gex_matrix)

# assigning human names to pc matrix
principle_components <- principle_components[row.names(human_gex_matrix),]

# Ensuring equal rows, equals 3000
ncol(t(principle_components))
nrow(human_gex_matrix)

# Multiplies the transpose of the principal component matrix by the human gex matrix
multiplication <- t(principle_components) %*% human_gex_matrix

# Creates new dimensional reduction on human Seurat object with the pc projections
human[["pca_projections"]] <- CreateDimReducObject(embeddings = t(multiplication), key = "pca_proj")
```

## Running GLM to determine significance of PCS
```{r}
# Add human cell embeddings to Seurat metadata
human <- AddMetaData(human, human@reductions$pca_projections@cell.embeddings, col.name = colnames(human@reductions$pca_projections@cell.embeddings))
```

```{r}
# Add dummy variable to classify cell as AD or C
human@meta.data$dv <- 0
Idents(human) <- human@meta.data$dv
human_ad <- subset(human, subset = Genotype == "AD")
Idents(human, cells = Cells(human_ad)) <- 1
human@meta.data$dv <- Idents(human)
Idents(human) <- human@meta.data$summary
table(human@meta.data$Genotype, human@meta.data$dv)
```


```{r}
# Save human object with pca_projection
#saveRDS(human, file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_pca.RData"))
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_pca.RData"))
```

```{r}
# Run GLM with all PCs
ad.model <- glm(dv ~ pcaproj_1 + pcaproj_2 + pcaproj_3 + pcaproj_4 + pcaproj_5 + 
                  pcaproj_6 + pcaproj_7 + pcaproj_8 + pcaproj_9 + pcaproj_10 +
                  pcaproj_11 + pcaproj_12 + pcaproj_13 + pcaproj_14 + pcaproj_15 +
                  pcaproj_16 + pcaproj_17 + pcaproj_18 + pcaproj_19 + pcaproj_20 +
                  pcaproj_21 + pcaproj_22 + pcaproj_23 + pcaproj_24 + pcaproj_25 +
                  pcaproj_26 + pcaproj_27 + pcaproj_28 + pcaproj_29 + pcaproj_30 +
                  pcaproj_31 + pcaproj_32 + pcaproj_33 + pcaproj_34 + pcaproj_35 +
                  pcaproj_36 + pcaproj_37 + pcaproj_38 + pcaproj_39 + pcaproj_40 +
                  pcaproj_41 + pcaproj_42 + pcaproj_43 + pcaproj_44 + pcaproj_45 +
                  pcaproj_46 + pcaproj_47 + pcaproj_48 + pcaproj_49 + pcaproj_50, data = human@meta.data, family = binomial)
```

```{r}
summary(ad.model)
```
AIC of 1424.8

```{r}
saveRDS(ad.model, file = paste0(home_dir, "/AD_human_objects_ortholog/pca_glm.RData"))
```

## RErunning GLM with significant PCs
```{r}
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_pca.RData"))
```

```{r}
# Run GLM
ad.model2 <- glm(dv ~ pcaproj_1 + pcaproj_2 + pcaproj_3 +
                   pcaproj_6 + pcaproj_9 + pcaproj_10 +
                   pcaproj_11 + pcaproj_12 + 
                   pcaproj_15 + pcaproj_16 +
                   pcaproj_19 +
                   pcaproj_23 + pcaproj_25 + pcaproj_28 + pcaproj_29 +
                   pcaproj_26 + pcaproj_31 + pcaproj_33 +
                   pcaproj_38 + pcaproj_39 +
                   pcaproj_41 +
                   pcaproj_46 + pcaproj_47 + pcaproj_48 + pcaproj_50, data = human@meta.data, family = binomial)

```

```{r}
summary(ad.model2)
```

AIC of 1399.7

```{r}
saveRDS(ad.model2, file = paste0(home_dir, "/AD_human_objects_ortholog/pca_glm2.RData"))
```

## RErunning GLM with highly significant PCs
```{r}
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_pca.RData"))
```


```{r}
# Run GLM
ad.model3 <- glm(dv ~ pcaproj_1 + pcaproj_3 +
                   pcaproj_6 + pcaproj_9 + pcaproj_10 +
                   pcaproj_11 + pcaproj_12 + pcaproj_15 +
                   pcaproj_16 +
                   pcaproj_33 + pcaproj_38 + 
                   pcaproj_46 + pcaproj_47
                  , data = human@meta.data, family = binomial)

```

```{r}
summary(ad.model3)
```

AIC of 1461.7

```{r}
saveRDS(ad.model3, file = paste0(home_dir, "/AD_human_objects_ortholog/pca_glm3.RData"))
```

## Comparing All GLMs
```{r}
ad.model1 <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/pca_glm.RData"))
ad.model2 <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/pca_glm2.RData"))
ad.model3 <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/pca_glm3.RData"))
models <- c(ad.model1, ad.model2, ad.model3)
```

```{r}
summary(ad.model1)
summary(ad.model2)
summary(ad.model3)
anova(ad.model1)
```

GLM with all significant PCs performs the best

## Visualize PCs
```{r}
# plot pc projections to visualize
for(i in 2:50){
p1 <- DimPlot(human, reduction = "pca_projections", dims = c(1, i), group.by = "Genotype")
ggsave(filename = paste0(home_dir, "/pc_plot_pca/pc_", i, ".pdf"))
}
```


```{r}
# Deviance barplot visualization
anova_df <- anova(ad.model2)
anova_df <- anova_df %>%
  arrange(desc(Deviance))
anova_df <- anova_df %>%
  filter(!row.names(anova_df) %in% c("NULL"))
anova_df

p2 <- ggplot(anova_df, aes(x = reorder(row.names(anova_df), Deviance), y = Deviance)) + 
  geom_bar(stat = "identity") + coord_flip() +
  scale_y_log10() + 
  labs(x = "Principal Component",
       title = "Significant PCA Loadings") +
   theme(axis.text.y = element_text(size = 10))

cairo_ps(filename= paste0(home_dir, "/Figures/Figure 3/3C_pca.eps"), width = 9, height = 7) # name the file as an EPS
p2 #variable of the figure that is defined/saved, can be used any name
dev.off()
```



```{r}
sessionInfo()
```

