---
title: "AD Ortholog Exploration"
author: "Alexander Bergendorf"
date: "2023-08-07"
output: html_document
---


# Library Import
```{r}
library(Seurat)
library(mixOmics)
library(h2o)
library(tidyverse)
library(orthogene)
library(dittoSeq)
library(elasticnet)
home_dir <- "C:/Users/aberg/OneDrive/Desktop/Brubaker Summer Research 2023"
```

# sPCA exploration
## Loading Mouse Data
```{r}
combined <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_adwt_labeled.RData"))
combined <- subset(combined, subset = summary == "Microglia")
```

```{r}
# Filtering mouse object based on good barcodes found
good_microglia <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/good_microglia_barcodes.RData"))
combined <- subset(combined, cells = good_microglia)
combined
```


```{r}
# Normalized and scale data, do not center - spca() from mixomics will center
DefaultAssay(combined) <- "RNA"
combined <- NormalizeData(combined)
all.genes <- rownames(combined)
combined <- ScaleData(combined, features = all.genes, do.center = FALSE)
```





## Run sPCA
```{r}
# Transpose mouse matrix and run spca, get 50 principal components
mouse_mat <- as.matrix(t(combined@assays$RNA@scale.data))
mouse_embeddings <- mixOmics::spca(mouse_mat, ncomp = 50, scale = FALSE, center = TRUE)
```

```{r}
# Save spca object output
#saveRDS(mouse_embeddings, file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_embeddings.RData"))
mouse_embeddings <-readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_embeddings.RData"))
```

```{r}
# Variance explained for spca pulled from spca object
variance_explained <- mouse_embeddings$prop_expl_var$X
variance_explained <- variance_explained /sum(variance_explained)
write.csv(variance_explained, paste0(home_dir, "/spca_variances_mouse.csv"))
```



```{r}
# Create spca reduction on mouse object and save
combined[["spca"]] <- CreateDimReducObject(embeddings = mouse_embeddings$x, loadings = mouse_embeddings$rotation, key = "spca")
saveRDS(combined, file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_spca_loadings.RData"))
#combined <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_spca_loadings.RData"))
```



```{r}
# Pull feature sparse principal component loadings from object
principle_components <- combined@reductions$spca@feature.loadings
head(principle_components)
principle_components <- as.data.frame(principle_components)
```



## Set Up Human Data
```{r}
# Load human data, pull out only good human microglia
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_adwt_labeled_ortholog.RData"))
microglia <- subset(human, subset = summary == "Microglia")
microglia
good_microglia <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/good_microglia_barcodes.RData"))
human <- subset(microglia, cells = good_microglia)
human
```



```{r}
# Normalize and Scale data, pull human gex_matrix
DefaultAssay(human) <- "RNA"
human <- NormalizeData(human)
all.genes <- rownames(human)
human <- ScaleData(human, features = all.genes)
human_gex_matrix <- as.data.frame(human@assays$RNA@scale.data)
```



```{r}
# Confirming only orthologs in GEX matrix
ortholog_mat <- convert_orthologs(gene_df = row.names(human_gex_matrix),
                                    input_species = "human",
                                    output_species = "mouse",
                                    non121_strategy = "drop_both_species")
# Filtering the orthologs based on the 15607 principal component loadings
ortholog_mat <- ortholog_mat %>%
  filter(row.names(ortholog_mat) %in% row.names(principle_components))
# Filter the human_gex_matrix based on the 15607 principal component loadings
human_gex_matrix <- human_gex_matrix %>%
  filter(row.names(human_gex_matrix) %in% ortholog_mat$input_gene)
# Ordering human_gex_matrix in the same manner as the pc loadings
row.names(human_gex_matrix) <- row.names(ortholog_mat)

# Arrange alphabetically 
principle_components  <- principle_components %>%
  arrange(row.names(principle_components))
# Arrange alphabetically 
human_gex_matrix <- human_gex_matrix %>%
  arrange(row.names(human_gex_matrix))
```




```{r}
# converting to matrix
principle_components <- as.matrix(principle_components)
human_gex_matrix <- as.matrix(human_gex_matrix)
# Ensuring equal rows, equals 3000
ncol(t(principle_components))
nrow(human_gex_matrix)
# assigning human names to pc matrix
principle_components <- principle_components[row.names(human_gex_matrix),]
# Multiplies the transpose of the principal component matrix by the human gex matrix
multiplication <- t(principle_components) %*% human_gex_matrix
# Creates new dimensional reduction on human Seurat object with the spc projections
human[["spca_projections"]] <- CreateDimReducObject(embeddings = t(multiplication), key = "spca_proj")
```


## Running GLM to determine significance of PCS
```{r}
# Add human cell embeddings to Seurat metadata
human <- AddMetaData(human, human@reductions$spca_projections@cell.embeddings, col.name = colnames(human@reductions$spca_projections@cell.embeddings))
```

```{r}
# Add dummy variable to classify cell as AD or C
human@meta.data$dv <- 0
Idents(human) <- human@meta.data$dv
human_ad <- subset(human, subset = Genotype == "AD")
Idents(human, cells = Cells(human_ad)) <- 1
human@meta.data$dv <- Idents(human)
Idents(human) <- human@meta.data$summary
table(human@meta.data$Genotype, human@meta.data$dv)
```


```{r}
#(human, file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_spca.RData"))\
human <- readRDS(paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_spca.RData"))
```


```{r}
# Run GLM with all PCs
ad.model <- glm(dv ~ spcaproj_1 + spcaproj_2 + spcaproj_3 + spcaproj_4 + spcaproj_5 + 
                  spcaproj_6 + spcaproj_7 + spcaproj_8 + spcaproj_9 + spcaproj_10 +
                  spcaproj_11 + spcaproj_12 + spcaproj_13 + spcaproj_14 + spcaproj_15 +
                  spcaproj_16 + spcaproj_17 + spcaproj_18 + spcaproj_19 + spcaproj_20 +
                  spcaproj_21 + spcaproj_22 + spcaproj_23 + spcaproj_24 + spcaproj_25 +
                  spcaproj_26 + spcaproj_27 + spcaproj_28 + spcaproj_29 + spcaproj_30 +
                  spcaproj_31 + spcaproj_32 + spcaproj_33 + spcaproj_34 + spcaproj_35 +
                  spcaproj_36 + spcaproj_37 + spcaproj_38 + spcaproj_39 + spcaproj_40 +
                  spcaproj_41 + spcaproj_42 + spcaproj_43 + spcaproj_44 + spcaproj_45 +
                  spcaproj_46 + spcaproj_47 + spcaproj_48 + spcaproj_49 + spcaproj_50, data = human@meta.data, family = binomial)
```

```{r}
summary(ad.model)
```



```{r}
saveRDS(ad.model, file = paste0(home_dir, "/AD_human_objects_ortholog/spca_glm.RData"))
```

## RErunning GLM with significant PCs
```{r}
#human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_spca.RData"))
```

2, 3, 4, 6, 7, 9, 10, 11, 12, 14, 15, 17, 20, 22, 23, 24, 25, 29, 30, 32, 36, 42, 45, 46, 47, 50
```{r}
# Run GLM
ad.model2 <- glm(dv ~ spcaproj_2 + spcaproj_3 + spcaproj_4 +
                   spcaproj_6 + spcaproj_7 + spcaproj_9 + spcaproj_10 +
                   spcaproj_11 + spcaproj_12 + spcaproj_14 + spcaproj_15 + 
                   spcaproj_17 + spcaproj_20 + 
                   spcaproj_22 + spcaproj_23 + spcaproj_24 + spcaproj_25 +
                   spcaproj_29 + spcaproj_30 +
                   spcaproj_32 + 
                   spcaproj_36 +
                   spcaproj_42 + spcaproj_45 +
                   spcaproj_46 + spcaproj_47 + spcaproj_50
                   ,
                  data = human@meta.data, family = binomial)
```

```{r}
summary(ad.model2)
```

```{r}
saveRDS(ad.model2, file = paste0(home_dir, "/AD_human_objects_ortholog/spca_glm2.RData"))
```

## RErunning GLM with Highly Significant PCs
```{r}
#human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_spca.RData"))
```

3, 6, 7, 9, 10, 11, 12, 14, 15, 20, 23, 25, 42, 46
```{r}
# Run GLM
ad.model3 <- glm(dv ~ spcaproj_3 +
                   spcaproj_6 + spcaproj_7 + spcaproj_9 + spcaproj_10 +
                   spcaproj_11 + spcaproj_12 + spcaproj_14 + spcaproj_15 + 
                   spcaproj_20 + 
                   spcaproj_23 + spcaproj_25 +
                   spcaproj_42 +
                   spcaproj_46
                 , data = human@meta.data, family = binomial)
```

```{r}
summary(ad.model3)
```

```{r}
saveRDS(ad.model3, file = paste0(home_dir, "/AD_human_objects_ortholog/spca_glm3.RData"))
```

## Comparing all GLMS
```{r}
ad.model1 <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/spca_glm.RData"))
ad.model2 <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/spca_glm2.RData"))
ad.model3 <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/spca_glm3.RData"))
models <- c(ad.model1, ad.model2, ad.model3)
```

```{r}
summary(ad.model1)
summary(ad.model2)
summary(ad.model3)
```

## Visualizing Highly Different PCs
```{r}
# plot pc projections to visualize
for(i in 2:50){
p1 <- DimPlot(human, reduction = "spca_projections", dims = c(1, i), group.by = "Genotype")
ggsave(filename = paste0(home_dir, "/pc_plot_spca/pc_", i, ".pdf"))
}
```


```{r}
# Deviance barplot visualization
anova_df <- anova(ad.model2)
anova_df <- anova_df %>%
  arrange(desc(Deviance))
anova_df <- anova_df %>%
  filter(!row.names(anova_df) %in% c("NULL"))
anova_df

p2 <- ggplot(anova_df, aes(x = reorder(row.names(anova_df), Deviance), y = Deviance)) + 
  geom_bar(stat = "identity") + coord_flip() +
  scale_y_log10() + 
  labs(x = "Principal Component",
       title = "Significant sPCA Loadings") + 
  theme(axis.text.y = element_text(size = 8))

cairo_ps(filename= paste0(home_dir, "/Figures/Figure 3/3C_spca.eps"), width = 9, height = 7) # name the file as an EPS
p2 #variable of the figure that is defined/saved, can be used any name
dev.off()
```



```{r}
sessionInfo()
```

