---
title: "PC Variance Explained"
author: "Alexander Bergendorf"
date: "2023-10-30"
output: html_document
---

# Library Import
```{r}
library(Seurat)
library(mixOmics)
library(h2o)
library(tidyverse)
library(orthogene)
library(dittoSeq)
library(openxlsx)
home_dir <- "C:/Users/aberg/OneDrive/Desktop/Brubaker Summer Research 2023"
```

# sPCA
## Data Import
```{r}
# Loading mouse object with spca loadings
mouse_spca <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_spca_loadings.RData"))

# Loading human object with spca projections
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_spca.RData"))
```



```{r}
# Get mouse feature loadings
mouse_loading_f <- mouse_spca@reductions$spca@feature.loadings
# Get human gene expression matrix
t_filt_human_ex <- t(human@assays$RNA@scale.data)
```


```{r}
# Get the variance of the spca projections
var_mh <- lapply(1:ncol(mouse_loading_f), function(x) (t(mouse_loading_f[,x])%*%t(t_filt_human_ex)%*%t_filt_human_ex%*%mouse_loading_f[,x])/(sum(diag(t(mouse_loading_f)%*%t(t_filt_human_ex)%*%t_filt_human_ex%*%mouse_loading_f))))
```



```{r}
# Reorganizing variance vector
variance_vec <- data.frame(variance = unlist(var_mh))
names_vec <- colnames(mouse_spca@reductions$spca@feature.loadings)
variance_vec$principal_component <- names_vec
write.csv(variance_vec, file = paste0(home_dir, "/spca_variances.csv"), row.names = FALSE)
#variance_vec <- read.csv(file = paste0(home_dir, "/spca_variances.csv"))
```



```{r}
# Variance explained of projections
p1 <- ggplot(variance_vec, aes(x = principal_component, y = variance)) + 
  geom_bar(stat = "identity") +
   theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    labs(x = "Principle Component",
       y = "Variance Explained",
       title = "Percent Variance Explained sPCA")

cairo_ps(filename= paste0(home_dir, "/Figures/Figure 3/3_variance_explained_spca.eps"), width = 7, height = 7) # name the file as an EPS
p1 #variable of the figure that is defined/saved, can be used any name
dev.off()
```



# PCA
## Data Import
```{r}
# Loading mouse object with pca loadings
mouse_pca <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_pca_loadings.RData"))

# loading human object with pca projections
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_mouse_proj_pca.RData"))
```

```{r}
# Getting mouse loadings
mouse_loading_f <- data.frame(mouse_pca@reductions$pca@feature.loadings)

# Getting human gene expression matrix
t_filt_human_ex <- data.frame(human@assays$RNA@scale.data)
```

```{r}
# Filtering human gene expression matrix to only include the 3000 genes in mouse matrix
ortholog_mat <- convert_orthologs(gene_df = row.names(t_filt_human_ex),
                                    input_species = "human",
                                    output_species = "mouse",
                                    non121_strategy = "drop_both_species")

ortholog_mat <- ortholog_mat %>%
  filter(row.names(ortholog_mat) %in% row.names(mouse_loading_f))
t_filt_human_ex <- t_filt_human_ex %>%
  filter(row.names(t_filt_human_ex) %in% ortholog_mat$input_gene)
row.names(t_filt_human_ex) <- row.names(ortholog_mat)
mouse_loading_f  <- mouse_loading_f %>% arrange(row.names(mouse_loading_f))
t_filt_human_ex <- t_filt_human_ex %>%
  arrange(row.names(t_filt_human_ex))
```

```{r}
# Get mouse feature loadings from principal component analysis
mouse_loading_f <- as.matrix(mouse_loading_f)
# get filtering human gene expression matrix
t_filt_human_ex <- t(as.matrix(t_filt_human_ex))
```

```{r}
# Calculate variances of human pca
var_mh <- lapply(1:ncol(mouse_loading_f), function(x) (t(mouse_loading_f[,x])%*%t(t_filt_human_ex)%*%t_filt_human_ex%*%mouse_loading_f[,x])/(sum(diag(t(mouse_loading_f)%*%t(t_filt_human_ex)%*%t_filt_human_ex%*%mouse_loading_f))))
```

```{r}
variance_vec <- data.frame(variance = unlist(var_mh))
names_vec <- colnames(mouse_pca@reductions$pca@feature.loadings)
variance_vec$principal_component <- names_vec
write.csv(variance_vec, file = paste0(home_dir, "/pca_variances.csv"))
#variance_vec <- read.csv(file = paste0(home_dir, "/pca_variances.csv"))
```


```{r}
# Percent variance explained
p2 <- ggplot(variance_vec, aes(x = principal_component, y = variance)) + 
  geom_bar(stat = "identity") +
   theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    labs(x = "Principle Component",
       y = "Variance Explained",
       title = "Percent Variance Explained PCA")

cairo_ps(filename= paste0(home_dir, "/Figures/Figure 3/3_variance_explained_pca.eps"), width = 7, height = 7) # name the file as an EPS
p2 #variable of the figure that is defined/saved, can be used any name
dev.off()
```


```{r}
sessionInfo()
```

