---
title: "AD Visualization"
author: "Alexander Bergendorf"
date: "2023-09-28"
output: html_document
---

# Library Import
```{r}
library(Seurat)
library(mixOmics)
library(h2o)
library(tidyverse)
library(orthogene)
library(dittoSeq)
library(openxlsx)
home_dir <- "C:/Users/aberg/OneDrive/Desktop/Brubaker Summer Research 2023"
```

# Data Import
```{r}
human <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/human_adwt_labeled_ortholog.RData"))
good_microglia <- readRDS(file = paste0(home_dir, "/AD_human_objects_ortholog/good_microglia_barcodes.RData"))
human <- subset(human, cells = good_microglia)
```

```{r}
human@meta.data
```


```{r}
DefaultAssay(human) <- "RNA"
human <- NormalizeData(human)
human <- FindVariableFeatures(human, selection.method = "vst", nfeatures = 3000)
all.genes <- rownames(human)
human <- ScaleData(human, features = all.genes)
human <- RunPCA(human, features = VariableFeatures(object = human))
human <- FindNeighbors(human, dims = 1:10)
human <- FindClusters(human, resolution = 0.5)
human <- RunUMAP(human, dims = 1:10)
```

```{r}
dittoDimPlot(human, "Genotype")
```


```{r}
mouse <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/mouse_adwt_labeled.RData"))
good_microglia <- readRDS(file = paste0(home_dir, "/AD_mouse_objects_ortholog/good_microglia_barcodes.RData"))
mouse <- subset(mouse, cells = good_microglia)
```

```{r}
mouse@meta.data
```


```{r}
DefaultAssay(mouse) <- "RNA"
mouse <- NormalizeData(mouse)
mouse <- FindVariableFeatures(mouse, selection.method = "vst", nfeatures = 3000)
all.genes <- rownames(mouse)
mouse <- ScaleData(mouse, features = all.genes)
mouse <- RunPCA(mouse, features = VariableFeatures(object = mouse))
mouse <- FindNeighbors(mouse, dims = 1:10)
mouse <- FindClusters(mouse, resolution = 0.5)
mouse <- RunUMAP(mouse, dims = 1:10)
```


```{r}
dittoDimPlot(mouse, "seurat_clusters")
```

```{r}
markers <- FindAllMarkers(mouse, assay = "RNA", only.pos = TRUE, logfc.threshold = 0.5, test.use = "MAST", latent.vars = "orig.ident") 

clust_counts <- table(mouse@meta.data[,"seurat_clusters"]) 

clust_markers_list <- list() 

for(i in unique(markers$cluster)){ 

clustname <- paste0('cluster', i, '_N', clust_counts[i]) 

clust_markers_list[[clustname]] <- subset(markers, markers$cluster == i) 

rownames(clust_markers_list[[clustname]]) <- clust_markers_list[[clustname]]$gene 

} 

clust_markers_list$cluster5_N49
```
# Cell Counts
```{r}
cell_counts <- read.csv(file = paste0(home_dir, "/Figures/Figure 1/cell_counts.csv"))
cell_counts
```

```{r}
cell_counts$Sample <- as.factor(cell_counts$Sample)
cell_counts
```


```{r}
levels(cell_counts$Sample) <- c("5XFAD", "AD", "WT", "Control")
p <- ggplot(data=cell_counts, aes(x=Cell_Type, y=Count, fill=Sample)) +
geom_bar(stat="identity", color="black", position=position_dodge())+
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust=1))

cairo_ps(filename= paste0(home_dir, "/Figures/Figure 1/1F.eps"), width = 9, height = 7) # name the file as an EPS
p #variable of the figure that is defined/saved, can be used any name
dev.off()
```

